environment/src/main/java/com/sequenceiq/environment/environment/validation/validators/EncryptionKeyArnValidator.java

```diff
@@ -102,7 +103,11 @@ public ValidationResult validate(EnvironmentValidationDto environmentValidationD
                     platformResources().encryptionKeys(extendedCloudCredential, region, Collections.emptyMap()));
             if (encryptionKeys.getCloudEncryptionKeys().stream().map(CloudEncryptionKey::getName)
                     .noneMatch(s -> s.equals(encryptionKeyArn))) {
-                validationResultBuilder.error("The provided encryption key does not exist in the given region's encryption key list for this credential.");
+                validationResultBuilder.error("Following encryption keys are retrieved from the cloud " +
+                                encryptionKeys.getCloudEncryptionKeys().stream().map(CloudEncryptionKey::getName).collect(Collectors.toList()) +
```

Comment:
I suggest extracting the whole `encryptionKeys.getCloudEncryptionKeys().stream().map(CloudEncryptionKey::getName).collect(Collectors.toList())` expression into a local variable; the same name mapping happens in the enclosing `if` condition, so it would be nice to avoid duplicating this.